vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -431,41 +511,39 @@ and
             let projectGuid = Guid(site.ProjectGuid)
             let projectFileName = site.ProjectFileName()
             let projectDisplayName = projectDisplayNameOf projectFileName
-
             let projectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectDisplayName)
+    
+            projectInfoManager.UpdateProjectInfo(tryGetOrCreateProjectId workspace, projectId, site, workspace, userOpName)
```

Comment:
This call has been moved from under the ``if ... then`` below.  I believe this causes it to be executed many more times in a DAG graph of projects + project references, since we recursively call ``setup`` for each referenced proejct

Is there any specific reason to make this change?  Is it because you call ``SetupProjectFile`` again for each change in the CPS project?  In that case, it feels like you should **only** be calling ``UpdateProjectInfo`` in the incremental case for a change to a CPS project - the rest of ``SetupProjectFile`` isn't needed

There's an exception to that - the calls to ``AddProjectReference`` are only being made in ``SetupProjectFile`` - but that is addressed in the as-yet-unmerged #3499  (with which this will have conflicts I think - hence trying to minimize the diff)